### PR TITLE
Toast: refactor 'button' prop to 'primaryAction' to enable upcoming redesign changes

### PR DESCRIPTION
**Pull request by AlbertCarreras**
## Breaking change

API change. From `button` (Node) to a restricted object `{ accessibilityLabel, href, label, onClick, rel, size, target }

Added a new undocumented prop `_dangerouslySetPrimaryAction`. Use `_dangerouslySetPrimaryAction` as an exception, only for legacy custom buttons components. 

Run the following codemod to detect button props and replace them manually:

```
yarn codemod detectManualReplacement ~/path/to/your/code
--component=Toast
--prop=button
```
 

#### What changed?

Toast: refactor 'button' prop to enable upcoming redesign changes

#### Why?

'button' prop doesn't allow to control button sizes. refactoring to 'primaryAction' to enable upcoming redesign changes

### Checklist

- [X] Added unit and Flow Tests
- [X] Added documentation + accessibility tests

### CODE REVIEW

File: packages/gestalt/src/Toast.js [hunk 1]
```diff
@@ -107,16 +124,28 @@ export default function Toast({
...
           </Flex.Item>
-
-          {button ? (
+          {primaryAction || _dangerouslySetPrimaryAction ? (
             // Allow button text to wrap on mobile
-            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>{button}</Flex.Item>
+            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+              {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
+              {!_dangerouslySetPrimaryAction &&
+              primaryAction?.accessibilityLabel &&
```
Comment (dangerismycat) [inline packages/gestalt/src/Toast.js:137]: I like that we're making these checks, but I think we should move them to ToastPrimaryAction, where we can throw an actual error. Flow will already be yelling at the developer about missing fields; throwing an error would accomplish the same thing you're doing here, but with more direct feedback about what's missing and how to fix it.

File: packages/gestalt/src/ToastPrimaryAction.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// @flow strict
+import { type Node } from 'react';
+import Button from './Button.js';
+import Link from './Link.js';
+
+type Props = {|
+  accessibilityLabel?: string,
```
Comment (dangerismycat) [inline packages/gestalt/src/ToastPrimaryAction.js:7]: `accessibilityLabel` and `label` should be required, no?

**Review by dangerismycat — APPROVED** [review]

File: packages/gestalt/src/Toast.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Toast.js:137]: I can set an error to double down on the user feedback, though Flow is already covering it

File: packages/gestalt/src/ToastPrimaryAction.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/ToastPrimaryAction.js]: DONE

**Pull request merged**